test/helper_test.rb

```diff
@@ -180,4 +180,16 @@ def test_vite_image_tag
         vite_image_tag('images/logo.png', srcset: { 'images/logo-2x.png' => '2x' }, alt: 'Logo')
     }
   end
+
+  if Rails.gem_version >= Gem::Version.new('7.1.0')
+    def test_vite_picture_tag
+      assert_equal <<~HTML.gsub(/\n\s*/, ''), vite_picture_tag('images/logo.svg', 'images/logo.png', class: 'test', image: { alt: 'Logo' })
+        <picture class="test">
+          <source srcset="/vite-production/assets/logo.322aae0c.svg" type="image/svg+xml" />
```

Comment:
I realize this PR is ~9 months old, but this line has different output for me.

Running Rails 7.1.3.4, the `type` attribute is different. I see a failure like:

```
Failure:
HelperTest#test_vite_picture_tag [/Users/mjankowski/repos/vite_ruby/test/helper_test.rb:186]
Minitest::Assertion: --- expected
+++ actual
@@ -1 +1 @@
-"<picture class=\"test\"><source srcset=\"/vite-production/assets/logo.322aae0c.svg\" type=\"image/svg+xml\" /><source srcset=\"/vite-production/assets/logo.f42fb7ea.png\" type=\"image/png\" /><img alt=\"Logo\" src=\"/vite-production/assets/logo.f42fb7ea.png\" /></picture>"
+"<picture class=\"test\"><source srcset=\"/vite-production/assets/logo.322aae0c.svg\" type=\"svg\" /><source srcset=\"/vite-production/assets/logo.f42fb7ea.png\" type=\"png\" /><img alt=\"Logo\" src=\"/vite-production/assets/logo.f42fb7ea.png\" /></picture>"
```

Maybe related to https://github.com/rails/rails/pull/48266 ?

Curious to see a CI run after https://github.com/ElMassimo/vite_ruby/pull/467

Reply:
@mjankowski I can't reproduce what you are seeing locally. Using vite_ruby @ main, and running `bundle up` to get the latest gems, the test still passes for me:

```
$ bundle exec rake test TEST=test/helper_test.rb
...
14 tests, 36 assertions, 0 failures, 0 errors, 0 skips
```

```
$ bundle show rails
/Users/mbrictson/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/rails-7.1.3.4
```

---

Digging through the Rails source, my guess is that `ActionView::Template::SimpleType` is somehow being used in your environment, causing the mime type lookup to return `svg` instead of the correct `image/svg+xml`.

I found this comment:

> SimpleType is mostly just a stub implementation for when Action View is used without Action Dispatch.

So maybe Action Dispatch isn't being loading in your environment, for some reason?

https://github.com/rails/rails/blob/v7.1.3.4/actionview/lib/action_view/template/types.rb#L9

Reply:
Ah, thanks ... think I tracked it down.

I have a branch going trying to get rid of some of deprecation noise during spec run, and I did a bit of branch jam-up on myself, and I think I had `eager_load` disabled during that failure, which may explain the lack of the template being present (and it falling back to just ext string) during the run.